config.go

```diff
@@ -549,6 +551,63 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
+// getGithubAppAuth returns app id and app private key if it exists.
+// returns false if it does not have both of those fields.
+func (s *Settings) getGithubAppAuth() (int64, []byte, bool) {
+	key, ok := s.Expansions[GithubAppPrivateKey]
+	if s.AuthConfig.Github != nil && s.AuthConfig.Github.AppId != 0 && ok && key != "" {
+		return s.AuthConfig.Github.AppId, []byte(key), true
+	}
+	return 0, nil, false
+}
+
+func (s *Settings) GetGithubAppClient(caller, owner, repo string, retry utility.HTTPRetryFunction, delay utility.HTTPDelayFunction) *http.Client {
+	appId, key, exists := s.getGithubAppAuth()
+	if !exists {
+		return nil
+	}
+	itr, err := ghinstallation.NewAppsTransport(http.DefaultTransport, appId, key)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "failed to create github jwt app transport",
+			"app_id":  appId,
+		}))
+		return nil
```

Comment:
Should this really squash the error rather than return it? The preference is to return errors instead of `grip.Error`ing in such low-level infrastructure code, unless there's good reason for it (or it's too diffcult to change existing code). This log statement doesn't seem very traceable, and I don't think anyone would know to look for it.